alert.go

```diff
@@ -563,6 +588,18 @@ func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map
 			continue
 		} else if pass {
 			return AlertLevel(l)
+		} else {
+			lrse := a.levelResets[l]
```

Comment:
Doesn't the determineLevel function need to know the current level? This logic doesn't seem right but maybe I am misunderstanding the purpose of the reset expressions.

My thinking was that if an alert enters a state with a reset expression it can't leave that state until the reset function is false. Meaning the determineLevel would check for any reset expressions for the current level first and only evaluate other expressions if the reset expression is false.

Maybe it would help to write up the tests at this point so we can agree on the expected behavior before we dig into the implementation details too far.